### PR TITLE
Replace loadimpact with k6io in github.com/loadimpact/k6

### LINKED ISSUE #1921: Moving the repo to github.com/k6io
As we mentioned in the [k6 v0.31.0 release notes](https://github.com/loadimpact/k6/blob/master/release%20notes/v0.31.0.md#planned-future-breaking-changes), we're going to move the k6 repo to the [k6io GitHub organization](https://github.com/k6io) as a part of the re-branding of Load Impact to k6. We've already moved all of the other repos, k6 is the last one, and its move will probably happen some time this week.

The actual move will happen in 2 steps:
- moving the repo from the GitHub UI - this should be transparent and without issues, since GitHub automatically redirects all links from the old repo address to the new one
- ~changing the Go package names from `github.com/loadimpact/k6/...` to `github.com/k6io/k6/...`~ update, see https://github.com/k6io/k6/issues/1921#issuecomment-811072303, we decided to go with vanity URLs like `go.k6.io/k6/...`  - this will likely break all Go apps that depend on k6 as a library when they update their used k6 versions. This includes currently open PRs and https://github.com/k6io/xk6, for which we'll release a new version that only works with new corresponding new k6 versions. Because of go modules and the github redirects, old k6 versions should still continue to work fine with the old module paths.

### PR DESCRIPTION
**Pull request by MStoykov**
fix #1921

### CODE REVIEW

**Comment by MStoykov** [comment]
I basically renamed it in go.mod and then `rg 'github.com/loadimpact/k6' -l -0|xargs -0 sed -i 's@github.com/loadimpact/k6@github.com/k6io/k6@g'`. 

Some fixes were required where I opened 3 so goimports does it thing

**Comment by MStoykov** [comment]
The latest changes work with https://github.com/k6io/xk6/pull/16 locally. 

But I can't figure out a combination of `replace` that makes it work so :shrug:

File: lib/old_archive_test.go [hunk 1]
```diff
@@ -93,14 +93,14 @@ func TestOldArchive(t *testing.T) {
 			metadata := `{"filename": "` + filename + `", "options": {}}`
 			fs := makeMemMapFs(t, map[string][]byte{
 				// files
-				"/files/github.com/loadimpact/k6/samples/example.js": []byte(`github file`),
+				"/files/github.com/k6io/k6/samples/example.js": []byte(`github file`),
```
Comment (na--) [inline lib/old_archive_test.go:96]: Fix the indentation in this file please

**Review by na-- — COMMENTED** [review]
Mostly LGTM, besides the one minor inline nitpick I mentioned inline. And we should first merge https://github.com/k6io/xk6/pull/16 and then re-run the CI, just to see if everything is fixed.

Another thing I am not sure about is whether we should touch the URLs in the release notes. I think I prefer not to, to leave them with `https://github.com/loadimpact/k6`. Given the flawless redirection, I'd prefer to not touch these files and muddle their git history unnecessarily. Having `loadimpact` in their URLs would also reflect that, at the time of the particular release, we were using the old repo path, which I think is better than essentially rewriting history.

**Comment by MStoykov** [comment]
> Another thing I am not sure about is whether we should touch the URLs in the release notes. I think I prefer not to, to leave them with https://github.com/loadimpact/k6. Given the flawless redirection, I'd prefer to not touch these files and muddle their git history unnecessarily. Having loadimpact in their URLs would also reflect that, at the time of the particular release, we were using the old repo path, which I think is better than essentially rewriting history.

I also wondered about that, but I have no idea if the redirect will work forever or if in a year or something it will be disabled :shrug:

**Comment by na--** [comment]
It should work. And we can always fix the URLs later, if it breaks for some reason... :smile:

File: lib/old_archive_test.go (hunk 1 above)
Comment (MStoykov) [reply lib/old_archive_test.go:96]: I was really hoping those will be caught by the golanci-lint, but apparently because there were other `gofumpt` reports it just decided this is not new :facepalm:

Comment (MStoykov) [reply lib/old_archive_test.go:96]: fixed in 5bb4f49 along others

**Review by na-- — APPROVED** [review]
LGTM now, though please squash everything into a single commit, so that the release notes history isn't affected

**Comment by MStoykov** [comment]
I am just going to squash all of them here in the interface later when goproxy catches on that `master` has updated ... 
see https://github.com/k6io/k6/pull/1936/commits/908f50b6a4472ee2917d03ae52f13e3b5ca07de2

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -43,15 +43,15 @@ To get a basic development environment for Go and k6 up and running, first make
...
 ```bash
-go get github.com/loadimpact/k6
+go get github.com/k6io/k6
 ```
 This will also build a `k6` binary and put it in `$GOPATH/bin`.
 
 **Building from source**:
 
-Standing in the repo root (`$GOPATH/src/github.com/loadimpact/k6`) you can build a k6 binary from source by running:
+Standing in the repo root (`$GOPATH/src/github.com/k6io/k6`) you can build a k6 binary from source by running:
```
Comment (imiric) [inline CONTRIBUTING.md:52]: Not for this PR, but should we remove the `GOPATH` details from here? And from the `Dockerfile` and `README` too?

File: README.md [hunk 3]
```diff
@@ -4,16 +4,16 @@
...
-  <a href="https://circleci.com/gh/loadimpact/k6/tree/master"><img src="https://img.shields.io/circleci/project/github/loadimpact/k6/master.svg" alt="Build status"></a>
-  <a href="https://goreportcard.com/report/github.com/loadimpact/k6"><img src="https://goreportcard.com/badge/github.com/loadimpact/k6" alt="Go Report Card"></a>
-  <a href="https://codecov.io/gh/loadimpact/k6"><img src="https://img.shields.io/codecov/c/github/loadimpact/k6/master.svg" alt="Codecov branch"></a>
+  <a href="https://github.com/k6io/k6/releases"><img src="https://img.shields.io/github/release/k6io/k6.svg" alt="Github release"></a>
+  <a href="https://circleci.com/gh/k6io/k6/tree/master"><img src="https://img.shields.io/circleci/project/github/k6io/k6/master.svg" alt="Build status"></a>
+  <a href="https://goreportcard.com/report/github.com/k6io/k6"><img src="https://goreportcard.com/badge/github.com/k6io/k6" alt="Go Report Card"></a>
+  <a href="https://codecov.io/gh/k6io/k6"><img src="https://img.shields.io/codecov/c/github/k6io/k6/master.svg" alt="Codecov branch"></a>
```
Comment (imiric) [inline README.md:10]: Some of these badges don't work yet, do we have to make some manual changes for that?

Also, we should replace the CircleCI one with a GH Actions badge. But also not part of this PR.

**Review by imiric — APPROVED** [review]
LGTM

I still wish we would've gone for a vanity URL that's under our control instead of continuing to depend on GitHub. It's a trivial amount of infrastructure to maintain to make it work last time I set it up, and `go.k6.io/k6` would be so clean. :)

File: CONTRIBUTING.md (hunk 2 above)
Comment (MStoykov) [reply CONTRIBUTING.md:52]: yeah, probably good idea, but not in this PR

File: README.md (hunk 3 above)
Comment (MStoykov) [reply README.md:10]: for me only the circleci one doesn't work, which should be dropped

**Review by na-- — CHANGES_REQUESTED** [review]
do not merge yet please

**Comment by na--** [comment]
Closing this because we decided to go with vanity URLs - https://github.com/k6io/k6/pull/1935
